src/components/timeout-dialog/timeout-dialog.js

```diff
@@ -76,8 +77,26 @@ function TimeoutDialog($module) {
     validateInput(options);
     settings = mergeOptionsWithDefaults(options, localisedDefaults);
     setupDialogTimer();
+    broadcastSessionActivity();
+    listenForSessionActivityAndResetDialogTimer();
   }
 
+  const broadcastSessionActivity = () => {
+    if (sessionActivityChannel) {
+      sessionActivityChannel.postMessage(getDateNow());
+    }
+  };
+
+  const listenForSessionActivityAndResetDialogTimer = () => {
+    if (sessionActivityChannel) {
+      sessionActivityChannel.onmessage = (event) => {
+        const timeOfActivity = event.data;
+        cleanup();
+        setupDialogTimer(timeOfActivity);
```

Comment:
passing the timestamp from the fg to the bg tab in case the bg tab for some reason doesn't immediately get this message, maybe because the tabs js is suspended somehow